staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -163,6 +163,13 @@ const (
 	// Deprecates and removes SelfLink from ObjectMeta and ListMeta.
 	RemoveSelfLink featuregate.Feature = "RemoveSelfLink"
 
+	// owner: @serathius
+	// beta: v1.30
+	//
+	// Allow watch cache to create a watch on a dedicated RPC.
+	// This prevents watch cache from being starved by other watches.
+	SeparateCacheWatchRPC featuregate.Feature = "SeparateCacheWatchRPC"
```

Comment:
This is desired - we're bug fixing.
The idea behind a feature-gate here is purely to give a mechanism for disabling this without changing the code.

Reply:
Is there any situation we know of where this change can work worse than the existing code? If so, I would expect the gate to be off in the backport (similar to https://github.com/kubernetes/kubernetes/pull/121196)

Reply:
xref https://github.com/kubernetes/kubernetes/pull/123532#discussion_r1504871843

> _If_ we backport this (I'm on the fence there as well since this isn't a regression), we could default the gate off in backports as well to make the change opt-in

Reply:
> Is there any situation we know of where this change can work worse than the existing code?

Nothing that I'm aware of - @serathius - anything you can add?

Reply:
I'm not aware of any regressions, however I only tested it's behavior on high traffic watch, haven't qualified it on more broad set of test.